### PR TITLE
fix: disconnect previous wallet

### PR DESCRIPTION
**Pull request by schmanu**
## What it solves
When switching the wallet the previously connected wallet stays connected.
This can lead to many edge cases for the social signer.

## How this PR fixes it
If the user successfully switches the wallet, we disconnect the old wallet label.

## How to test it
1. Login with google
2. Click switch wallet and connect to Metamask
3. Disconnect Metamask
4. you are no longer connected to google automatically

## Checklist
* [x] I've tested the branch on mobile 📱
* [x] I've documented how it affects the analytics (if at all) 📊
* [x] I've written a unit/e2e test for it (if applicable) 🧑‍💻

### CODE REVIEW

**Comment by katspaugh** [comment]
That was actually a feature. If we're throwing it away, you can just remove the Switch wallet button altogether.

**Comment by usame-algan** [comment]
I would also be in favour of limiting this to social login only and adding tracking for the Switch Wallet button to decide later if we should remove it or not.

File: src/hooks/wallets/useOnboard.ts [hunk 1]
```diff
@@ -130,8 +131,19 @@ export const connectWallet = async (
...
+  const newWallets = await connectWallet(onboard)
+  const newWalletLabel = newWallets ? getConnectedWallet(newWallets)?.label : undefined
+
+  // If the wallet actually changed we disconnect the old connected wallet.
+  if (!newWalletLabel || oldWalletLabel !== ONBOARD_MPC_MODULE_LABEL) {
+    return
+  }
+
+  if (newWalletLabel !== oldWalletLabel) {
+    await onboard.disconnectWallet({ label: oldWalletLabel })
```
Comment (katspaugh) [inline src/hooks/wallets/useOnboard.ts:145]: So we dusconnect if the old wallet is social login? What if the new wallet is social login and the old one is MM? Will it work fine?

Comment (schmanu) [reply src/hooks/wallets/useOnboard.ts:145]: Yes exactly. 
Only the Social signer will be disconnected if switched to another wallet.
We could also remove this behavior once we support every chain. But this is currently causing issues QA found.

**Review by usame-algan — APPROVED** [review]
